sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1822,10 +1831,12 @@ pi_result piDeviceGetInfo(pi_device Device, pi_device_info ParamName,
   case PI_DEVICE_INFO_PARTITION_PROPERTIES: {
     // SYCL spec says: if this SYCL device cannot be partitioned into at least
     // two sub devices then the returned vector must be empty.
-    uint32_t ZeSubDeviceCount = 0;
-    ZE_CALL(zeDeviceGetSubDevices, (ZeDevice, &ZeSubDeviceCount, nullptr));
-    if (ZeSubDeviceCount < 2) {
-      return ReturnValue(pi_device_partition_property{0});
+    if (!Device->isSubDevice()) {
```

Comment:
You are going to return non-zero vector for sub-devices, why is it OK? Also please change the code to get # of sub-devices from the cache (instead of repeating call to zeDeviceGetSubDevices)